client/client-core/src/main/java/software/amazon/smithy/java/client/core/Client.java

```diff
@@ -80,12 +80,16 @@ protected <I extends SerializableStruct, O extends SerializableStruct> Completab
         IdentityResolvers callIdentityResolvers = identityResolvers;
         ClientInterceptor callInterceptor = interceptor;
 
-        // First apply overrides from interceptors.
-        ClientConfig callConfig = callInterceptor.modifyBeforeCall(new CallHook<>(operation, config, input));
-        // Overrides given per/operation take precedence over interceptors.
+        //If there is an override config first apply that before sending to interceptors.
+        ClientConfig callConfig = config;
```

Comment:
@mtdowling and I discussed this on the original CR. I thought it would be confusing if request overrides specified at an individual call site could be trumped by global overrides, so override configs should have highest precedence. Why does it need to be changed?

Reply:
Modified to now give the interceptor a view of the overriden config but also again override the config.